lighthouse-core/test/gather/gather-runner-test.js

```diff
@@ -27,7 +27,7 @@ jest.mock('../../lib/stack-collector.js', () => () => Promise.resolve([]));
  * @param {(...args: TParams) => TReturn} fn
  */
 function makeParamsOptional(fn) {
-  return /** @type {(...args: RecursivePartial<TParams>) => TReturn} */ (fn);
+  return /** @type {(...args: Nullable<RecursivePartial<TParams>>) => TReturn} */ (fn);
```

Comment:
This function is now so super-extra-optional that I question the value of type checking its uses, but w/e :P

Reply:
I think i'm missing something.. how does adding `null` to the type of each parameter item harm the type checking? the nullable bit isn't not recursive:

![image](https://user-images.githubusercontent.com/4071474/88433393-94fac280-cdc3-11ea-8c3d-7e673bed738f.png)

Reply:
> I think i'm missing something.. how does adding null to the type of each parameter item harm the type checking?

That question appears to answer itself :P

Reply:
If it's that apparent @brendankenny would you mind still explaining to us plebeians? I have the same question as @connorjclark 😆 

why did we need to make individual options nullable but not in a recursive way? did ts really treat `fn(null)` as passing a `fn(arg: number | undefined)` signature before 3.9.7?

Reply:
> If it's that apparent @brendankenny would you mind still explaining to us plebeians? I have the same question as @connorjclark 😆

That part was a bit too flippant, my apologies :) I meant in the literal sense that when a type is already a union of a couple of things, and you throw more things in that union, the type may still be technically correct but is likely less and less useful (the limit being `any`).

Much more constructively, now 😛...

> why did we need to make individual options nullable but not in a recursive way? did ts really treat `fn(null)` as passing a `fn(arg: number | undefined)` signature before 3.9.7?

It looks like yes, but only in limited circumstances (hence the change being limited to basically this file). `Config` [allows `null` for a lot of its properties](https://github.com/GoogleChrome/lighthouse/blob/8b4803d3c82ae679c5a62d8a30dbe98b8ed07162/lighthouse-core/config/config.js#L344-L351) (e.g. an `-A` config doesn't need `passes`), but we (usually) check for those before calling, which this test code isn't doing (like [here in `Runner`](https://github.com/GoogleChrome/lighthouse/blob/d8d8caa2b295b81cad83865621f8a789bab8a707/lighthouse-core/runner.js#L206-L208), love that old-school-LH phrasing that probably wouldn't be very helpful to an end user). So it's a legitimate complaint by tsc, even though we know all the test configs we're making have `passes`.

Pretty sure the typescript fix was https://github.com/microsoft/TypeScript/pull/37195; see that issue for details.

Buuuut, now that I've looked into this, pretty sure there's a much better fix than what I have here. I'll be in the shame cube :)